src/services/programService/httpClient.js

```diff
@@ -27,14 +27,108 @@ import fetch from 'node-fetch';
 import { PROGRAM_SERVICE_HTTP_ROOT } from '../../config';
 import { restErrorResponseHandler } from '../../utils/restUtils';
 
-const getProgramPublicFields = async (programShortName) => {
-	const url = `${PROGRAM_SERVICE_HTTP_ROOT}/public/program?name=${programShortName}`;
-	const response = await fetch(url, {
+//data formatters
+const formatPublicProgram = (program) => ({
+	name: program.name,
+	shortName: program.shortName,
+	description: program.description,
+	website: program.website,
+	institutions: program.programInstitutions?.map((institution) => institution.name) || [],
+	countries: program.programCountries?.map((country) => country.name) || [],
+	regions: program.processingRegions?.map((region) => region.name) || [],
+	cancerTypes: program.programCancers?.map((cancer) => cancer.name) || [],
+	primarySites: program.programPrimarySites?.map((primarySite) => primarySite.name) || [],
+});
+
+const formatPrivateProgram = (program) => {
+	return program.program;
+};
+
+const formatPrivateProgramList = (programList) => programList.map(formatPrivateProgram);
+
+const formatJoinProgramInvite = (invitation) => ({
+	...invitation,
+	createdAt: new Date(invitation.createdAt),
+	expiresAt: new Date(invitation.expiresAt),
+	acceptedAt: new Date(invitation.acceptedAt),
+	user: { ...invitation.user, role: invitation.user.role.value },
+	program: {
+		...invitation.program,
+		institutions: invitation.program.programInstitutions,
+		countries: invitation.program.programCountries,
+		cancerTypes: invitation.program.programCancers,
+		primarySite: invitation.program.programPrimarySites,
+	},
+});
+
+//private fields
+export const listPrivatePrograms = async (jwt = null) => {
+	const url = `${PROGRAM_SERVICE_HTTP_ROOT}/programs`;
+	return await fetch(url, {
+		method: 'get',
+		headers: {
+			Authorization: `Bearer ${jwt}`,
+		},
+	})
+		.then(restErrorResponseHandler)
```

Comment:
using this new error handler, `restErrorResponseHandler` after git merge develop.

Reply:
I don't understand this comment.

Reply:
You rewrote the error handler, I git pull in develop and is using the new handler you wrote.